### DISCUSSION TITLE
Insert/Invoke Widget within widget code

### DISCUSSION THREAD

**Discussion by welford** [TiddlyWiki/TiddlyWiki5#5799] [Q&A] [answered]
I have two widgets in the same file, when i render the second one i would like to insert the first, is this possible?

e.g. `<$widget2 a="123" b="456"/>`

I would like this to expand to

`<div>123<$widget1>456</$widget1></div>`

and then have $widget1 run, I assume that would happen via "Widget.prototype.render.call"

**Comment by saqimtiaz** [comment, marked as answer]
One option is to add a parse tree representation for widget1 into the children for widget2 in its `execute` method before calling `this.makeChildWidgets` (along with whatever parseTree will generate the content you want around widget1). This is a somewhat limited example: https://github.com/Jermolene/TiddlyWiki5/blob/2da7ae0b735c45725802d813d15d0c807f95e65b/core/modules/widgets/edit.js#L58-L62

Note that the "internals" plugin is very useful for typing content into a tiddler editor and using the preview option to see the resultant parse tree.

You absolutely can also create a widget1 instance and call render as well, but it will still need a parseTree.
If you ignore the context and just look at the code, this is an example of creating and rendering a tiddler widget in this manner: https://github.com/Jermolene/TiddlyWiki5/discussions/5787#discussioncomment-865675

**Reply by welford** [reply]
thanks Saq, I think your first suggestion is what i need - how cani generate a parsetreenode.children structure for an dom element i have created by myself?

**Reply by welford** [reply]
var wikiParser = $tw.wiki.parseText("text/vnd.tiddlywiki", myNewWidgetText, {parseAsInline: true});
Is what i am looking for.

Thanks Saq
